### PR TITLE
Test coverage

### PR DESCRIPTION
**Pull request by TingDaoK**
- add test coverage
- Updated the localhost for http/2 to be setup from builder instead.
- trivial updates for the localhost port to avoid conflict with other local server.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .builder/action/local-server-setup.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            return
+
+        base_dir = os.path.dirname(os.path.realpath(__file__))
+        dir = os.path.join(base_dir, "..", "..", "tests", "py_localhost")
+        os.chdir(dir)
+
+        p_server = subprocess.Popen([python_path, "server.py"])
+        p_non_tls_server = subprocess.Popen([python_path, "non_tls_server.py"])
+
+        print("something to make different")
```
Comment (waahm7) [inline .builder/action/local-server-setup.py:42]: I am not sure what is the purpose of this print. Maybe you forgot to remove this?

File: .builder/action/aws-c-http-test.py [hunk 2]
```diff
@@ -33,5 +33,7 @@ def run(self, env):
             os.chdir('../../aws-c-http')
 
         actions.append(['ctest', '--output-on-failure'])
+        # generate the test coverage report whne possible, will be ignored by ctest if there is no test coverage data available.
```
Comment (waahm7) [inline .builder/action/aws-c-http-test.py:36]: trivial: 
```suggestion
        # generate the test coverage report whenever possible, will be ignored by ctest if there is no test coverage data available.
```

File: .builder/action/local-server-setup.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    This action should be run in the 'pre_build_steps' or 'build_steps' stage.
+    """
+
+    def run(self, env):
+        self.env = env
+        python_path = sys.executable
+        # install dependency for mock server.
+        # Okay to fail, and if it faile, you will know when you enable the localhost test.
+        # We don't need it to success on every platform we have
```
Comment (waahm7) [inline .builder/action/local-server-setup.py:27]: Trivial: 
```suggestion
        # Install dependency for mock server.
        # Okay to fail, and if it fails, you will know when you enable the localhost tests.
        # We don't need it to succeed on every platform we have.
```

**Review by waahm7 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -155,28 +155,16 @@ jobs:
...
-    - name: Configure local host
-      run: |
-        python3 -m pip install h2
-        cd ./tests/py_localhost/
-        python3 server.py &
-        python3 non_tls_server.py &
     - name: Build and test
       run: |
         python3 -c "from urllib.request import urlretrieve; urlretrieve('${{ env.BUILDER_HOST }}/${{ env.BUILDER_SOURCE }}/${{ env.BUILDER_VERSION }}/builder.pyz?run=${{ env.RUN }}', 'builder.pyz')"
-        python3 builder.pyz build -p aws-c-http --cmake-extra=-DENABLE_LOCALHOST_INTEGRATION_TESTS=ON
```
Comment (graebm) [inline .github/workflows/ci.yml:167]: why doesn't `localhost-test-linux` require `-DENABLE_LOCALHOST_INTEGRATION_TESTS=ON` anymore?

that `ENABLE_LOCALHOST_INTEGRATION_TESTS` is still set in the `localhost-test-mac` and `localhost-test-win` runs

File: codebuild/linux-integration-tests.yml [hunk 5]
```diff
@@ -19,7 +19,7 @@ phases:
       - aws s3 cp s3://aws-crt-test-stuff/setup_proxy_test_env_h2.sh /tmp/setup_proxy_test_env.sh
       - chmod a+xr /tmp/setup_proxy_test_env.sh
       - python3 -c "from urllib.request import urlretrieve; urlretrieve('$BUILDER_HOST/$BUILDER_SOURCE/$BUILDER_VERSION/builder.pyz', 'builder.pyz')"
-      - python3 builder.pyz build -p aws-c-http --cmake-extra=-DENABLE_PROXY_INTEGRATION_TESTS=ON
+      - python3 builder.pyz build -p aws-c-http --cmake-extra=-DENABLE_PROXY_INTEGRATION_TESTS=ON --cmake-extra=-DENABLE_LOCALHOST_INTEGRATION_TESTS=ON
```
Comment (graebm) [inline codebuild/linux-integration-tests.yml:22]: why don't we run code coverage from codebuild, where we have the best test coverage?

File: .github/workflows/ci.yml (hunk 4 above)
Comment (TingDaoK) [reply .github/workflows/ci.yml]: was trying to find out why the test coverage report different lines covered.

**Review by graebm — APPROVED** [review]

**Pull request merged**